File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int maxE = nums[0];//first element can be include
        for(int i = 0; i< nums.length; i++){
            sum = sum + nums[i];
            //now check the sum is more then maxE or not
            if(sum > maxE){
                maxE = sum;
            }
            if(sum < 0){
                sum = 0;
            }
        }
        return maxE;
        
        
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n=(int)nums.size();
        int start=0,end=0,point=0,end_temp=0,start_temp=nums[0];
        
        for(int i=0;i<n;i++){
            end_temp+=nums[i];
            if(start_temp<end_temp){
                start_temp=end_temp;
                start=point;
                end=i;
            }
            if(end_temp<0){
                end_temp=0;
                point=i+1;
            }
        }
        return start_temp;
    }
};